src/extension/background-script/migrations/index.ts

```diff
@@ -0,0 +1,65 @@
+import type { Allowance } from "~/types";
+
+import db from "../db";
+import state from "../state";
+
+export type Migration = keyof typeof migrations;
+
+const shouldMigrate = (name: Migration): boolean => {
+  const { migrations } = state.getState();
+
+  // if migrations is blank
+  if (!migrations) {
+    return true;
+  }
+  return !migrations.includes(name);
+};
+
+const setMigrated = (name: Migration): Promise<void> => {
+  let { migrations } = state.getState();
+  if (!migrations) {
+    migrations = [];
+  }
+  migrations.push(name);
+  state.setState({
+    migrations: migrations,
+  });
+  return state.getState().saveToStorage();
+};
+
+const migrations = {
+  migrateisUsingLegacyLnurlAuthKeySetting: async () => {
+    const { settings } = state.getState();
+    const allowances = await db.allowances
+      .filter((allowance: Allowance) => {
+        return !!allowance.lnurlAuth;
+      })
+      .toArray();
+
+    // if there is an allowance that uses lnurlAuth we enable the legacy signing
+    if (allowances.length > 0) {
+      const newSettings = {
+        ...settings,
+        isUsingLegacyLnurlAuthKey: true,
+      };
+      state.setState({
+        settings: newSettings,
+      });
+      // state is saved with the setMigrated call
+    }
+  },
+};
+
+const migrate = async () => {
+  // going forward we can iterate through the the migrations object above and DRY this up:
+  // Object.keys(migrations).forEach((name: string) => {
+  if (shouldMigrate("migrateisUsingLegacyLnurlAuthKeySetting")) {
+    console.info(
+      "Running migration for: migrateisUsingLegacyLnurlAuthKeySetting"
+    );
+    await migrations["migrateisUsingLegacyLnurlAuthKeySetting"]();
+    await setMigrated("migrateisUsingLegacyLnurlAuthKeySetting");
```

Comment:
Many times the same-very-long-string. 

You could save it at top of the file as `const IS_USING_LEGACY = "migrateisUsingLegacyLnurlAuthKeySetting"`

and then use this const everyhwere:

```TS

const migrations = {
  [IS_USING_LEGACY]: async () => {
    const { settings } = state.getState();

 // .... 

  if (shouldMigrate(IS_USING_LEGACY)) {
    console.info(`Running migration for: ${IS_USING_LEGACY}`);
    await migrations[IS_USING_LEGACY]();
    await setMigrated(IS_USING_LEGACY);
  }
```

Reply:
I guess then I can also just use a shorter name? :)

Reply:
Currently it's ok to handle because it only accepts migration types. But we could have a look again once more migrations are needed.